nixos-module/default.nix

```diff
@@ -7,14 +7,22 @@
     ./boot-m1n1
   ];
 
-  config = {
-    hardware.asahi.pkgs = if config.hardware.asahi.pkgsSystem != "aarch64-linux"
-    then import (pkgs.path) {
-        system = config.hardware.asahi.pkgsSystem;
-        crossSystem.system = "aarch64-linux";
-      }
-    else pkgs;
-  };
+  config =
+    let
+      cfg = config.hardware.asahi;
+    in {
+      nixpkgs.overlays = lib.mkBefore [ cfg.overlay ];
+
+      hardware.asahi.pkgs =
+        if cfg.pkgsSystem != "aarch64-linux"
+        then
+          import (pkgs.path) {
+            crossSystem.system = "aarch64-linux";
+            localSystem.system = cfg.pkgsSystem;
+            overlays = [ cfg.overlay ];
+          }
+        else pkgs;
+    };
```

Comment:
Unsure if the `pkgsSystem` option below is actually working properly. Seemed to want to cross build the whole system instead of just the kernel in the installer, but it's possible I installed it wrong. I need to re-check this.

Reply:
Does it do this if you set `pkgsSystem` to `aarch64-linux`?

Reply:
That works fine. See the latest review.